marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -192,22 +192,42 @@ declare %private function hent:fix-options-exp($nodes as node()*)
               <search:facet-option>descending</search:facet-option>
             </search:range>
           </search:constraint>,
-          hent:fix-options-exp($n/node())
+          hent:fix-options-exp($n/node(), $sortable-properties)
+        }
+      case element(search:constraint) return
+        element { fn:node-name($n) } {
+          $n/@*,
+          let $constraint-name :=
+            for $attribute in $n/attribute()
+              where $attribute/name() = "name"
+              return $attribute
+          let $search-range-node := $n/search:range
+          return if (fn:empty($search-range-node) or fn:not(map:contains($sortable-properties, xs:string($constraint-name)))) then
+            hent:fix-options-exp($n/node(), $sortable-properties)
+            else
+              element {fn:node-name($search-range-node)} {
+                for $attribute in $search-range-node/attribute()
+                  let $name := $attribute/name()
```

Comment:
For this, you can do $search-range-node/*[local-name(.) != "facet"] instead of a for loop

Reply:
`$search-range-node/*[local-name(.) != "facet"]` is returning the child elements. But, `$search-range-node/attribute()[not(name()='facet')]` is returning the attributes of the element search-range-node. Replaced for loop with this xpath